docs/source/configuration/slots.md

```diff
@@ -0,0 +1,295 @@
+---
+myst:
+  html_meta:
+    "description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:description": "Slots are insertion points in the Volto rendering tree structure."
+    "property=og:title": "Slots"
+    "keywords": "Volto, Plone, frontend, React, configuration, slots, viewlets"
+---
+
+# Slots
+
+Slots provide a way for Volto add-ons to insert their own components at predefined locations in the rendered page.
+
+```{note}
+This concept is inspired by the Plone Classic UI {doc}`plone:classic-ui/viewlets`.
+```
+
+
+## Anatomy
+
+Slots are named, and they can contain a list of different slot components.
+Slot components are also named, and they are registered in the {ref}`configuration registry using a specific API for slots <configuration-registry-for-slot-components>`.
+
+The main trait of a slot component is that its renderer is controlled by a list of conditions called {term}`predicates`.
+Multiple slot components can be registered under the same name, as long as they have different predicates.
+
+The following tree structure diagram illustrates these concepts.
+
+```text
+Slot (`toolbar`)
+├── SlotComponent (`edit`)
+│   ├── predicate (only appear in `/de/about`)
+│   ├── predicate (only appear if the content type is either a `Document` or `News Item`)
+│   └── no predicate (default when all predicates return `false`)
+├── SlotComponent (`contents`)
+└── SlotComponent (`more`)
+```
+
+At the root of the tree, there is a slot named `toolbar`.
+It contains three slot components, named `edit`, `contents`, and `more`.
+`edit` contains three slot components, two of which have a predicate and a third without a predicate.
```

Comment:
@sneridagh @stevepiercy I'm still confused by this tree and the description of a slot component that contains 3 slot components, since I don't see a notion of SlotComponents containing SlotComponents in the actual API. It's more that there are SlotComponents which are grouped by name before deciding which one wins for each name using the predicates, right?

Would it help and be accurate to show it like this instead?
```text
Slot (`toolbar`)
├── name=`edit`
│   ├── SlotComponent with predicate (only appear in `/de/about`)
│   ├── SlotComponent with predicate (only appear if the content type is either a `Document` or `News Item`)
│   └── SlotComponent with no predicate (default when all predicates return `false`)
├── name=`contents`
│   └── SlotComponent
└── name=`more`
    └── SlotComponent
```

Reply:
I agree the naming is a bit confusing. First the Slot (toolbar) that contains (named) slot components.
So you can register as many slot components under the same named slot component ('edit'). These registrations can have different predicates or none at all.
Maybe the last two components should also expose a real case, so they are more consistent?

Reply:
@davisagli how ZCA name such things? When you register a named view under other interfaces?

Reply:
@sneridagh I would explain it like this: In each slot, multiple slot components can be registered. Each slot component has a name. If there are multiple slot components registered with the same name, then the predicates are evaluated to decide which slot component will be rendered. The first slot component with a matching predicate will be used.

Reply:
Sounds great!

Reply:
Using our example, if what @davisagli wrote is true, then there would be a total of 5 registered SlotComponents. Is that correct? I tried [reading the code](https://github.com/plone/volto/pull/5775/files#diff-5e23fa7ab384750c59928974f281315a01f49d7e8751645b5a253fae32cf7890R236-R296), and it appears to be correct, but I'm not certain due to my lack of fluency in TypeScript.

If so, then our tree diagram should be flattened from 3 levels to 2, as well.